tests/test_google_ads_interrupted_sync_add_stream.py

```diff
@@ -0,0 +1,269 @@
+        return_value = {
+            'start_date':   '2021-12-01T00:00:00Z',
+            'user_id':      'not used?', # TODO ?
+            'conversion_window': '1',  # days
+            'customer_ids': ','.join(self.get_customer_ids()),
+            'login_customer_ids': [{"customerId": os.getenv('TAP_GOOGLE_ADS_CUSTOMER_ID'),
+                                    "loginCustomerId": os.getenv('TAP_GOOGLE_ADS_LOGIN_CUSTOMER_ID'),}],
+
+        }
+
+        # TODO_TDL-17911 Add a test around conversion_window_days
+        if original:
+            return return_value
+
+        return_value["start_date"] = self.start_date
+        return return_value
+
+
+    def assertIsDateFormat(self, value, str_format): # TODO needed in this test?
+        """
+        Assertion Method that verifies a string value is a formatted datetime with
+        the specified format.
+        """
+        try:
+            _ = dt.strptime(value, str_format)
+        except ValueError as err:
+            raise AssertionError(
+                f"Value does not conform to expected format: {str_format}"
+            ) from err
+
+
+    def test_run(self):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `currently_syncing`.
+                  The next sync job kicks off and the tap picks back up on that `currently_syncing` stream.
+
+        Expected State Structure:
+            state = {'currently_syncing': ('<stream-name2>', '<customer-id>'),
+            'bookmarks': {
+                '<stream-name1>': {'<customer-id>': {'<replication-key>': <completed-bookmark-value>}},
+                '<stream-name2>': {'<customer-id>': {'<replication-key>': <incomplete-bookmark-value>}},
+
+        Test Cases:
+         - Verify behavior is consistent when an added stream is selected between initial and resuming sync
+         - Verify an interrupted sync can resume based on the `currently_syncing` and stream level bookmark value
+         - Verify only records with replication-key values greater than or equal to the stream level bookmark are replicated on the resuming sync for the interrupted stream
+         - Verify the yet-to-be-synced streams are replicated following the interrupted stream in the resuming sync. All yet-to-be-synced streams must replicate before streams that were already synced.
+        """
+        print("Interrupted Sync Test for tap-google-ads with added stream")
+
+        # the following streams are under test as they all have 4 consecutive days with records e.g.
+        # ('2022-01-23T00:00:00.000000Z', '2022-01-23T00:00:00.000000Z', '2022-01-24T00:00:00.000000Z', '2022-01-25T00:00:00.000000Z')])}
+        streams_under_test = {'account_performance_report',
+                              'search_query_performance_report',
+                              'user_location_performance_report',
+        }
+
+        # Create connection using a recent start date
+        self.start_date = '2022-01-22T00:00:00Z'
```

Comment:
This can be updated in the override for  get_propertties. Then original_props does not need to be False on next line.